tf_quant_finance/black_scholes/asian_prices_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for asian_prices."""
+
+import functools
+
+from absl.testing import parameterized
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+
+import tf_quant_finance as tff
+from tensorflow.python.framework import test_util  # pylint: disable=g-direct-tensorflow-import
+
+
+@test_util.run_all_in_graph_and_eager_modes
+class AsianPriceTest(parameterized.TestCase, tf.test.TestCase):
+  """Tests for methods for the asian pricing module"""
+
+  def test_option_prices(self):
+    """Tests for methods for the asian pricing module. Results come from
+    'Implementing Derivatives Model' by Clewlow, Strickland p.118-123, via the
+    QuantLib test suite
+    """
+    spots = np.array([100.0])
+    dividend_rates = np.array([0.03])
+    discount_rates = np.array([0.06])
+    volatilities = np.array([0.2])
+    strikes = np.array([100.0])
+    expiries = 1.0
+    sampling_times = np.array([np.linspace(0, 1, 11)[1:]]).transpose()
+
+    expected_price = np.array([5.3425606635])
```

Comment:
could you please add a comment where this expected price is coming from?

Reply:
There is a comment in line 33-35 of this test that has the source of this price. Unfortunately I don't have access to the textbook it references, I've actually taken this test from QuantLib, here: https://github.com/lballabio/QuantLib/blob/7f69bb6fbe4ac76f0262b717cbf97c6dcd7b69d9/test-suite/asianoptions.cpp#L374

Most of the papers that do discuss geometric options unfortunately don't have sample prices for them. Instead, I've added a new test that compares the analytic pricing to a monte carlo calculation.